stdlib/public/core/IntegerParsing.swift

```diff
@@ -2,23 +2,166 @@
 //
 // This source file is part of the Swift.org open source project
 //
-// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
 // Licensed under Apache License v2.0 with Runtime Library Exception
 //
 // See https://swift.org/LICENSE.txt for license information
 // See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 //
 //===----------------------------------------------------------------------===//
 
+@_alwaysEmitIntoClient
+internal func _parseASCIIDigits<Result: FixedWidthInteger>(
+  _ codeUnits: UnsafeBufferPointer<UInt8>, radix: Int, isNegative: Bool
```

Comment:
Would it be possible to make this generic? Same with `_parseASCII`.

Then the implementation could just be:

```swift
public init?<S: StringProtocol>(_ text: S, radix: Int = 10) {
  // ... preconditions, etc
  text.utf8.withContiguousStorageIfAvailable { 
    _parseASCII($0, radix: radix)
  } ?? _parseASCII(text.utf8, radix: radix) // <- here. Parse the string's non-contiguous UTF8 directly.
}
```

From what I can see, the current implementation copies non-contiguous UTF8 in to a new `String` (line 78) - but this parsing function only visits each code-unit once, so I doubt there is any performance advantage to making the string contiguous.

Making the function generic would also help in situations where we have UTF8 code-units which are not literally stored in a `String`. I have some use-cases for this and would love to expose it as a public API in the future.

Reply:
That's the current design, and the cause of the underlying problem with code size. The purpose of the exercise here is to make these functions _not_ generic as much as possible.

Reply:
Hmm... I still think that having a feature to parse an integer from a generic collection of UTF8/ASCII bytes is valuable and it would be disappointing if users had to roll their own or convert to String/contiguous memory.

That said, I struggle to believe that the standard library really needs 20KB for all its uses. Perhaps the problem is just the inlining in to every call-site? What's the effect of just replacing `@inline(__always)` with `@inlinable @inline(__never)`? Apologies if its already mentioned above - there are lots of results and it's hard to correlate which each of them means.

Reply:
Yes, even with that.